--- a/contents/config/main.xml
+++ b/contents/config/main.xml
@@ -6,6 +6,10 @@
   <kcfgfile name=""/>
 
   <group name="General">
+    <entry name="hideDesktopWidgets" type="Bool" hidden="true">
+      <default>false</default>
+    </entry>
+
     <entry name="icon" type="String">
       <label>The name of the custom icon to use for the compact representation (e.g. on a small panel) of the Folder View applet. Only used if useCustomIcon is true.</label>
       <default>folder-symbolic</default>
--- a/contents/ui/main.qml
+++ b/contents/ui/main.qml
@@ -318,6 +318,16 @@
             appletContainerComponent: ContainmentLayoutManager.BasicAppletContainer {
                 id: appletContainer
 
+                opacity: Plasmoid.configuration.hideDesktopWidgets ? 0 : 1
+                visible: opacity > 0
+
+                Behavior on opacity {
+                    NumberAnimation {
+                        duration: Kirigami.Units.longDuration
+                        easing.type: Easing.InOutQuad
+                    }
+                }
+
                 editModeCondition: Plasmoid.immutable
                     ? ContainmentLayoutManager.ItemContainer.Locked
                     : ContainmentLayoutManager.ItemContainer.AfterPressAndHold
